framework/src/node/consensus/certificate_generation/commit_pool.ts

```diff
@@ -72,9 +72,10 @@ export class CommitPool {
 		clearInterval(this._jobIntervalID);
 	}
 
-	public addCommit(commit: SingleCommit): void {
-		if (!this._nonGossipedCommits.exists(commit)) {
-			this._nonGossipedCommits.add(commit);
+	public addCommit(commit: SingleCommit, local = false): void {
+		if (!this._nonGossipedCommits.exists(commit) && !this._nonGossipedCommitsLocal.exists(commit)) {
+			// eslint-disable-next-line @typescript-eslint/no-unused-expressions
```

Comment:
it seems like this expression work, but since eslint is complaining, maybe better to just switch with standard if condition